ush/remap.sh

```diff
@@ -0,0 +1,376 @@
+#   $2 - The path to the input netCDF file.
+#   $3 - The interpolation method to use (e.g., "remapbil").
+#   $4 - Comma-delimited rotation angles or trigonometric functions
+#        (e.g., "theta" or "cosang,sinang").
+#
+# Global Variables:
+#   output_path - The path to the output netCDF file.
+#
+# Example usage:
+#   cdo_rotate "u,v" "input_file.nc" "remapbil" "theta"
+#
+# This example rotates the "u" and "v" vector components using a
+# single angle variable "theta" and appends the rotated vectors to the
+# output netCDF file.
+#
+# Vector Rotation Options:
+#   nangles = 1:
+#      Rotate the specificed vectors accordingly; here it is assumed
+#      that a single angle (e.g., `theta`) defines the grid rotation
+#      angle; the units of `theta` are assumed to be radians.
+#
+#   nangles = 2:
+#     Rotate the specified vectors accordingly; here it is assumed
+#     that the `cos(theta)` and `sin(theta)` have been computed
+#     a'priori and assigned the input file variable names defined by
+#     `cosang` and `sinang` respectively.
+#
+function cdo_rotate(){    
+    local varnames="${1}"
+    local varfile="${2}"
+    local interp_type="${3}"
+    local angles="${4}"  
+    local var_rotate_path="${PWD}/rotate.nc"    
+
+    _comma_split_string "${varnames}"    
+    varname_array=("${global_array[@]}")    
+    _comma_split_string "${angles}"
+    angle_array=("${global_array[@]}")
+    _strip_whitespace "${varname_array[0]}"
+    xvar="${out_string}"
+    _strip_whitespace "${varname_array[1]}"
+    yvar="${out_string}"
+    nangles="${#angle_array[@]}"
+    
+    echo "Rotating and remapping variables ${xvar} and ${yvar} from file ${varfile}."
+    if [[ "${nangles}" == 1 ]]; then
+	_strip_whitespace "${angle_array[0]}"
+	theta="${out_string}"
+	cdo -expr,"xr=${xvar}*cos(${theta})-${yvar}*sin(${theta}); yr=${xvar}*sin(${theta})+${yvar}*cos(${theta})" -selname,"${xvar}","${yvar}","${theta}" "${varfile}" "${var_rotate_path}"
+    elif [[ "${nangles}" == 2 ]]; then
+	_strip_whitespace "${angle_array[0]}"
+	cosang="${out_string}"
+	_strip_whitespace "${angle_array[1]}"
+	sinang="${out_string}"
+	cdo -expr,"xr=${xvar}*${cosang}-${yvar}*${sinang}; yr=${xvar}*${sinang}+${yvar}*${cosang}" -selname,"${xvar}","${yvar}","${cosang}","${sinang}" "${varfile}" "${var_rotate_path}"
+	
+    else
+	echo "Vector rotations with ${nangles} attributes is not supported. Aborting!!!"
+	exit 102
+    fi
+
+    cdo_remap "xr" "${var_rotate_path}" "${interp_type}"
+    varname_update "xr" "${xvar}" "${output_path}"
+    cdo_remap "yr" "${var_rotate_path}" "${interp_type}"
+    varname_update "yr" "${yvar}" "${output_path}"
+    rm "${var_rotate_path}" >> /dev/null
+}
+
+#######
+
+# varname_update - Rename a variable in a netCDF file and write to a new file.
+#
+# Description:
+#   This function renames a variable in a netCDF file from
+#   `old_varname` to `new_varname` and writes the result to a new
+#   netCDF file specified by `ncfile`.
```

Comment:
Is this really a new file, or renaming within the same file? Seems like the latter since no old file is specified.

Reply:
It is the same file. Only `old_varname` is renamed to `new_varname`.

Reply:
Okay, just update the comment then.

Reply:
Done.

Reply:
Short description (just above this) still says "new file".

Reply:
Sorry. I missed this. Done.